pkg/runtime/framework/plugins/coscheduling/coscheduling_test.go

```diff
@@ -0,0 +1,326 @@
+		gocmp.Comparer(utiltesting.MPISecretDataComparer),
+	}
+	errorGetPodGroup := errors.New("failed to get PodGroup from API during Build")
+
+	cases := map[string]struct {
+		info                    *runtime.Info
+		trainJob                *trainerv1alpha1.TrainJob
+		objs                    []client.Object
+		wantInfo                *runtime.Info
+		wantObjs                []apiruntime.Object
+		wantPodGroupPolicyError error
+		wantBuildError          error
+	}{
+		"no action when info is nil": {},
+		"no action when podGroupPolicy is nil": {
+			info: &runtime.Info{
+				Labels: map[string]string{"key": "value"},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: nil,
+				},
+			},
+			trainJob: utiltesting.MakeTrainJobWrapper(metav1.NamespaceDefault, "trainJob").
+				UID("trainJob").
+				Trainer(
+					utiltesting.MakeTrainJobTrainerWrapper().
+						NumNodes(2).
+						Obj()).
+				Obj(),
+			wantInfo: &runtime.Info{
+				Labels: map[string]string{"key": "value"},
+			},
+		},
+		"no action when coscheduling is nil": {
+			info: &runtime.Info{
+				Scheduler: &runtime.Scheduler{},
+				Labels:    map[string]string{"key": "value"},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainerv1alpha1.PodGroupPolicy{
+						PodGroupPolicySource: trainerv1alpha1.PodGroupPolicySource{
+							Coscheduling: nil,
+						},
+					},
+				},
+			},
+			trainJob: utiltesting.MakeTrainJobWrapper(metav1.NamespaceDefault, "trainJob").
+				UID("trainJob").
+				Trainer(
+					utiltesting.MakeTrainJobTrainerWrapper().
+						NumNodes(2).
+						Obj()).
+				Obj(),
+			wantInfo: &runtime.Info{
+				Scheduler: &runtime.Scheduler{
+					PodLabels: map[string]string{
+						"scheduling.x-k8s.io/pod-group": "trainJob",
+					},
+				},
+				Labels: map[string]string{"key": "value"},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainerv1alpha1.PodGroupPolicy{},
+				},
+			},
+		},
+		"no action when trainJob is nil": {
+			info: &runtime.Info{
+				Scheduler: &runtime.Scheduler{},
+				Labels:    map[string]string{"key": "value"},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: nil,
+				},
+			},
+			trainJob: nil,
+			wantInfo: &runtime.Info{
+				Labels:    map[string]string{"key": "value"},
+				Scheduler: &runtime.Scheduler{},
+			},
+		},
+		"succeeded to build PodGroup": {
+			info: &runtime.Info{
+				Scheduler: &runtime.Scheduler{},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainerv1alpha1.PodGroupPolicy{
+						PodGroupPolicySource: trainerv1alpha1.PodGroupPolicySource{
+							Coscheduling: &trainerv1alpha1.CoschedulingPodGroupPolicySource{
+								ScheduleTimeoutSeconds: ptr.To[int32](30),
+							},
+						},
+					},
+				},
+				TemplateSpec: runtime.TemplateSpec{
+					PodSets: []runtime.PodSet{
+						{
+							Name:  "node",
+							Count: ptr.To[int32](1),
+						},
+					},
```

Comment:
Please add test case when multiple PodSet is in Info (e.g. Dataset Initializer + Node)